corehq/apps/data_interfaces/forms.py

```diff
@@ -567,19 +610,34 @@ def clean_property_match_definitions(self):
                     'property_value': property_value,
                     'match_type': match_type,
                 })
-
         return result
 
     def clean_filter_on_closed_parent(self):
         return true_or_false(self.cleaned_data.get('filter_on_closed_parent'))
 
-    def save_criteria(self, rule):
+    def clean_location_filter_definition(self):
+        value = self.cleaned_data.get('location_filter_definition')
+        try:
+            value = json.loads(value)
+        except (TypeError, ValueError):
+            self._json_fail_hard()
+
+        if value:
+            location_def = value[0]
+            if not location_def['include_child_locations']:
```

Comment:
Do you need to handle `KeyError`?

Reply:
Good point.

I'm thinking along the lines of simply doing this: 
```suggestion
            if not location_def.get('include_child_locations'):
```